### PR TITLE
Start Using BeginSignIn command and remove obsolete API command calls (Take 2)

### PR DESCRIPTION
**Pull request by yuwenmemon**
@robertjchen please review

Second try of https://github.com/Expensify/App/pull/10222

### Details
Start using the `BeginSignIn` command added in https://github.com/Expensify/Web-Expensify/pull/34408

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/218743

### Tests
1. Do a normal sign-in flow for a user that already has a validated account, make sure it works properly:
https://user-images.githubusercontent.com/4741899/182488142-8044224b-396a-46b2-97a2-69599c0ee76d.mp4

---

1. With the same account do the same sign in. Click "Forgot?" on the password form.
2. Make sure you see the following:
Screen Shot 2022-08-02 at 4 03 52 PM

3. And that you get send a password reset link email.
4. Click the "Resend link" button.
5. Make sure you get another email

---

1. Sign in with an email that DOES NOT already have an account
2. Make sure that right away you see:
Screen Shot 2022-08-02 at 4 14 01 PM

3. ...And that you get an email with a sign-in link:
Screen Shot 2022-08-02 at 4 14 25 PM

4. Before validating that account by clicking the link in the email, sign-in with that account again, make sure that you get an email with a sign-in link again.

---

1. Sign-in with a phone number that does not have an account yet, make sure you see the following:
Screen Shot 2022-08-02 at 4 16 39 PM

2. And that you get a text message with your magic sign-in link:
IMG_8397

---

**Error handling:**

1. Sign in with a number that is not actually a phone number, make sure you get the following error message:
Screen Shot 2022-08-08 at 2 51 13 PM

2. Sign in with a login that is not actually an email, make sure you get the following error message:
Screen Shot 2022-08-08 at 2 51 49 PM

### CODE REVIEW

**Review by robertjchen — APPROVED** [review]
LGTM! 👍

File: src/libs/actions/Session/index.js [hunk 1]
```diff
@@ -137,45 +102,42 @@ function resendValidationLink(login = credentials.login) {
...
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+                errors: {
+                    [DateUtils.getMicroseconds()]: 'Cannot get account details, please try again',
```
Comment (iwiznia) [inline src/libs/actions/Session/index.js:134]: Just saw this randomly. This is wrong, this is a string that should be localized. Can you send a PR to fix that please?

Comment (yuwenmemon) [reply src/libs/actions/Session/index.js:134]: Do you have a link for the context for this? These updates to `errors` have been breakneck - I want to make sure I have everything straight.

Comment (yuwenmemon) [reply src/libs/actions/Session/index.js:134]: Oh, wait I see it. Yeah, my bad, updating!

**Pull request merged**